File: src/main/java/io/mfedirko/contactme/ContactForm.java
package io.mfedirko.contactme;

import io.mfedirko.common.validation.ValidReCaptcha;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import lombok.Builder;
import lombok.Getter;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;
import org.hibernate.validator.constraints.Length;

import java.beans.ConstructorProperties;

@Getter
@Builder
@Value
@Jacksonized
public class ContactForm {
    @ValidReCaptcha
    String recaptcha;

    @NotEmpty
    @Length(min = 5, max = 50)
    String fullName;

    @Length(max = 50)
    @NotEmpty
    @Email
    String email;

    @NotEmpty
    @Length(max = 300)
    String messageBody;

    @ConstructorProperties({"g-recaptcha-response", "fullName", "email", "messageBody"})
    public ContactForm(String recaptcha, String fullName, String email, String messageBody) {
        this.recaptcha = recaptcha;
        this.fullName = fullName;
        this.email = email;
        this.messageBody = messageBody;
    }
}

File: src/main/kotlin/io/mfedirko/contactme/ContactForm.kt
package io.mfedirko.contactme

import io.mfedirko.common.validation.ValidReCaptcha
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotEmpty
import org.hibernate.validator.constraints.Length
import java.beans.ConstructorProperties

class ContactForm
@ConstructorProperties("g-recaptcha-response", "fullName", "email", "messageBody")
constructor(
    @field:ValidReCaptcha
    var recaptcha: String?,

    @field:NotEmpty
    @field:Length(min = 5, max = 50)
    var fullName: String?,

    @field:NotEmpty
    @field:Length(max = 50)
    @field:Email
    var email: String?,

    @field:NotEmpty
    @field:Length(max = 300)
    var messageBody: String?
)